### PR TITLE
Port to OSRM 5.0 API

### PR DESCRIPTION
**Pull request by lbud**

### CODE REVIEW

**Comment by karenzshea** [comment]
@lbud I'm hitting a bug with bearings requests somewhere around https://github.com/Project-OSRM/node-osrm/blob/c40d4776dfb266efd5143d81e5dd76015673048d/src/node_osrm.cpp#L213. It's not handling `null` bearing parameters properly.

Request format I think should be valid, `{ bearings: [null, [100, 20] null] }`

**Comment by lbud** [comment]
@karenzshea :bow: good catch, fixed in https://github.com/Project-OSRM/node-osrm/commit/aa451e5471056374fe23a53599e5d2315519b4b0 — this build is going to be broken until the OSRM lng/lat switch is fixed upstream, so I can't publish a binary quite yet.

File: bootstrap.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ CURRENT_DIR=$(pwd)
 # default to clang
 CXX=${CXX:-clang++}
 TARGET=${TARGET:-Release}
-OSRM_RELEASE=${OSRM_RELEASE:-"develop"}
+OSRM_RELEASE=${OSRM_RELEASE:-"rewrite/new-api"}
```
Comment (freenerd) [inline bootstrap.sh]: @lbud do you think we can get @MoKob's guidance work into the pre-release as well? He has this PR up https://github.com/Project-OSRM/osrm-backend/pull/2007 which he rebases every day against the `rewrite/new-api` branch by @TheMarex

Comment (lbud) [inline bootstrap.sh]: :grin: yes @freenerd as soon as I get this damn travis segfault sorted I will pull that in

**Comment by willwhite** [comment]
Just noticed that the prerelease tags look like `v0.5.0-prerelease` and `v0.5.0-prerelease.1`. Should these be `v5.0.0` instead of `v0.5.0`?

File: src/node_osrm.cpp [hunk 2]
```diff
@@ -375,257 +756,220 @@ struct RunQueryBaton
...
+void Engine::ParseResult(const osrm::engine::Status result_status_code, osrm::json::Object result)
+{
+    const auto message_iter = result.values.find("status_message");
+    const auto end_iter = result.values.end();
+    const auto result_code = static_cast<int>(result_status_code);
+
+    // 4xx : Invalid request
+    // 207 : No route found
+    // 208 : No edge found
+    if (result_code / 100 == 4 || result_code == 207 || result_code == 208)
```
Comment (TheMarex) [inline src/node_osrm.cpp]: 207 and 208 don't exist anymore, you can probably simplify here to a `== 400` check.

**Comment by TheMarex** [comment]
This branch is dead - current work happens in https://github.com/Project-OSRM/node-osrm/pull/161 and will be forced pushed here once guidance is merged.

**Comment by karenzshea** [comment]
With `v5.0.0-newapi.2`, errors are no longer being returned as expected.

I'm seeing in places that used to return proper `NoRoute` or `NoSegment` codes are now just returning errors like `[Tue, 08 Mar 2016 16:33:12 GMT] [error] [default] Error: Could not find a matching segment for coordinate 2`

**Comment by TheMarex** [comment]
:warning: Rebased on `develop` to get the git sha printed in every build.

**Comment by TheMarex** [comment]
Okay squashed and rebased most of the changes.

File: package.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
   "url": "https://github.com/Project-OSRM/node-osrm",
   "homepage": "http://project-osrm.org",
   "author": "Dane Springmeyer <springmeyer>",
-  "version": "4.9.0-develop.0",
+  "version": "develop",
   "osrm_release" : "develop",
```
Comment (springmeyer) [inline package.json]: Should this be `"version":"4.9.0"` @TheMarex ?

**Pull request merged**